src/views/PositionPlot/PositionPlotView.tsx

```diff
@@ -0,0 +1,147 @@
+import { matrix, multiply } from 'mathjs'
+import React, { FunctionComponent, useCallback, useMemo } from 'react'
+import colorForUnitId from 'views/common/colorForUnitId'
+import TimeScrollView, { computePanelDimensions, computePixelsPerSecond, get1dTimeToPixelMatrix, TimeScrollViewPanel } from '../RasterPlot/TimeScrollView/TimeScrollView'
+import { PositionPlotViewData } from './PositionPlotViewData'
+
+type Props = {
+    data: PositionPlotViewData
+    width: number
+    height: number
+}
+
+type PanelProps = {
+    pixelZero: number
+    dimensions: {
+        dimensionIndex: number
+        dimensionLabel: string
+        pixelTimes: number[]
+        pixelValues: number[]
+    }[]
+}
+
+const margins = {
+    left: 30,
+    right: 20,
+    top: 20,
+    bottom: 50
+}
+
+const panelSpacing = 4
+
+const PositionPlotView: FunctionComponent<Props> = ({data, width, height}) => {
+    const {startTimeSec, endTimeSec} = useMemo(() => {
+        const startTimeSec = min(data.timestamps)
+        const endTimeSec = max(data.timestamps)
+        return {startTimeSec, endTimeSec}
+    }, [data.timestamps])
+
+    // Compute the per-panel pixel drawing area dimensions.
+    const panelCount = 1
+    const { panelWidth, panelHeight } = useMemo(() => computePanelDimensions(width, height, panelCount, panelSpacing, margins), [width, height, panelCount])
+    const pixelsPerSecond = useMemo(() => computePixelsPerSecond(panelWidth, startTimeSec, endTimeSec), [endTimeSec, startTimeSec, panelWidth])
+
+    // We need to have the panelHeight before we can use it in the paint function.
+    // By using a callback, we avoid having to complicate the props passed to the painting function; it doesn't make a big difference
+    // but simplifies the prop list a bit.
+    const paintPanel = useCallback((context: CanvasRenderingContext2D, props: PanelProps) => {
+        context.strokeStyle = 'black'
+        context.setLineDash([5, 15]);
+        context.beginPath()
+        context.moveTo(0, props.pixelZero)
+        context.lineTo(width, props.pixelZero)
+        context.stroke()
+        context.setLineDash([]);
+
+        for (let dim of props.dimensions) {
+            context.strokeStyle = colorForUnitId(dim.dimensionIndex)
+            context.beginPath()
+            let first = true
+            for (let i=0; i<dim.pixelTimes.length; i++) {
+                const x = dim.pixelTimes[i]
+                const y = dim.pixelValues[i]
+                if (first) context.moveTo(x, y)
+                else context.lineTo(x, y)
+                first = false
```

Comment:
Might be easier to just set the initial position outside the loop as `context.moveTo(dim.pixelTimes[0], dim.pixelValues[0])`.

Reply:
In some ways that's better (avoids the if), but it requires keeping 2 lines of code in sync (what if the definition of x,y changes). Also we get an extra segment in the path - I can't think of how this would harm us, but still. Leaving it as is for now.